### PR TITLE
Feature engineering slides

### PR DESCRIPTION
**Pull request by topepo**

### CODE REVIEW

**Comment by topepo** [comment]
Once merged, I'll compile for the `2023-nyr` branch and commit the intermediate files

File: slides/advanced-01-feature-engineering.qmd
Comment (topepo) [inline slides/advanced-01-feature-engineering.qmd:702]: I'll merge the existing PRs ass-is (once review is done) and then go back and add tuning for these two pre-processing steps.

File: slides/advanced-01-feature-engineering.qmd
Comment (topepo) [inline slides/advanced-01-feature-engineering.qmd:518]: I'll make "extra" slides using some of these slides plus the effect encoding steps and workflow sets

File: slides/advanced-01-feature-engineering.qmd
Comment (EmilHvitfeldt) [inline slides/advanced-01-feature-engineering.qmd:676]: while this is true, the hashing function used in feature hashing are not cryptologically safe, rather they are fast and uniform.

**Review by EmilHvitfeldt — APPROVED** [review]
looking good, few notes

File: slides/advanced-01-feature-engineering.qmd
Comment (topepo) [reply slides/advanced-01-feature-engineering.qmd:676]: Good point.

File: slides/advanced-01-feature-engineering.qmd
Comment (hfrick) [inline slides/advanced-01-feature-engineering.qmd:479]: The learners will not necessarily have made their own workflow at this point. Are they supposed to use the one from the previous slides or pick a (different) model to make their own workflow?

File: slides/advanced-01-feature-engineering.qmd
Comment (hfrick) [inline slides/advanced-01-feature-engineering.qmd:593]:
```suggestion
There is a recipe step that will redefine factor levels based on their frequency in the training set: 
```

File: slides/advanced-01-feature-engineering.qmd
Comment (hfrick) [inline slides/advanced-01-feature-engineering.qmd:597]:
```suggestion
#| code-line-numbers: "4"
```

File: slides/advanced-01-feature-engineering.qmd
Comment (hfrick) [inline slides/advanced-01-feature-engineering.qmd:645]: The code output does not actually show the MAE (and the RMSE is worse).

File: slides/advanced-01-feature-engineering.qmd
Comment (hfrick) [inline slides/advanced-01-feature-engineering.qmd:655]:
```suggestion
Feature hashing (for more see [_FES_](https://bookdown.org/max/FES/encoding-predictors-with-many-categories.html), [_SMLTAR_](https://smltar.com/mlregression.html#case-study-feature-hashing), and [_TMwR_](https://www.tmwr.org/categorical.html#feature-hashing)):  
```

File: slides/advanced-01-feature-engineering.qmd
Comment (hfrick) [inline slides/advanced-01-feature-engineering.qmd:659]: this is a bit of mix between singular and plural, would this be correct:
```suggestion
- uses the character values of the levels 
- converts them to integer hash values
- uses the integers to assign them to a specific indicator column. 
```

File: slides/advanced-01-feature-engineering.qmd
Comment (hfrick) [inline slides/advanced-01-feature-engineering.qmd:672]:
```suggestion
# For "Max_Kuhn" put a '1' in column: 
```

File: slides/advanced-01-feature-engineering.qmd
Comment (hfrick) [inline slides/advanced-01-feature-engineering.qmd:682]:
```suggestion
- There is no real logic behind which factor levels are combined. 
```

File: slides/advanced-01-feature-engineering.qmd
Comment (hfrick) [inline slides/advanced-01-feature-engineering.qmd:679]: Maybe group the pros and cons?

File: slides/advanced-01-feature-engineering.qmd
Comment (hfrick) [inline slides/advanced-01-feature-engineering.qmd:686]: This doesn't really feel like a pro or con

File: slides/advanced-01-feature-engineering.qmd
Comment (hfrick) [inline slides/advanced-01-feature-engineering.qmd:690]: Let's add the hex for textrecipes!

File: slides/advanced-01-feature-engineering.qmd
Comment (hfrick) [inline slides/advanced-01-feature-engineering.qmd:704]:
```suggestion
  textrecipes::step_dummy_hash(agent) %>%
  textrecipes::step_dummy_hash(company) %>%
```
like we did earlier with `embed::step_umap()`

File: slides/advanced-01-feature-engineering.qmd
Comment (hfrick) [inline slides/advanced-01-feature-engineering.qmd:751]: Is this intentional?

File: slides/advanced-01-feature-engineering.qmd
Comment (hfrick) [inline slides/advanced-01-feature-engineering.qmd:591]: From here on out we don't have hexes in the corner anymore

**Review by hfrick — APPROVED** [review]
Nice!

File: slides/advanced-01-feature-engineering.qmd
Comment (topepo) [reply slides/advanced-01-feature-engineering.qmd:479]: Maybe I don't understand; we did [make one on the previous slide](https://github.com/tidymodels/workshops/pull/127/files#diff-757f60aaa2bcbca2701235fb4c5e1184fc52674ac8a7d67ea7e79fd5838fcd4bR462)

File: slides/advanced-01-feature-engineering.qmd
Comment (topepo) [reply slides/advanced-01-feature-engineering.qmd:686]: It's a pro since the signs make it slightly more possible to differentiate between confounded levels. I'll add a note.

File: slides/advanced-01-feature-engineering.qmd
Comment (topepo) [reply slides/advanced-01-feature-engineering.qmd:751]: Yes. It is like that in the previous versions (I think because there is a lot of potential output).

**Pull request merged**
